pandas/tests/io/formats/test_to_csv.py

```diff
@@ -223,3 +227,34 @@ def test_to_csv_multi_index(self):
 
         exp = "foo\nbar\n1\n"
         assert df.to_csv(index=False) == exp
+
+    @pytest.mark.parametrize('filename,compression,compressor', [
+        ("test.gz", "gzip", gzip.GzipFile),
+        ("test.gz", "infer", gzip.GzipFile),
+        ("test.bz2", "bz2", bz2.BZ2File),
+        ("test.bz2", "infer", bz2.BZ2File)
+    ])
+    def test_to_csv_compression(self, filename, compression, compressor):
+        # see GH issue 15008
+
+        df = DataFrame([1])
+        exp = "".join((",0", os.linesep, "0,1", os.linesep)).encode("ascii")
```

Comment:
why are we encoding as ascii?

Reply:
I need the result to be bytes not string, and in a python 2 and 3 compatible way. `os.linesep` is string (bytes) in python 2 and string (unicode) in python 3.